# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCSSFrameConstructor.h
# Commit: 2d16e9e90401
# Full Hash: 2d16e9e904010ba3389aad9729bac4f3744a588d
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2021-01-30 05:43:48
# Regressor Bug: 1683748
# File Overlap Count: 3
# Description:
#   Bug 1683748 - Support Grid/Flex/Table/Column layout for the rendered legend of a fieldset.  r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101555
# ==============================================================================

diff -r 5b99d9a7ad8d -r 2d16e9e90401 layout/base/nsCSSFrameConstructor.h
--- a/layout/base/nsCSSFrameConstructor.h	Fri Jan 29 11:30:44 2021 +0000
+++ b/layout/base/nsCSSFrameConstructor.h	Fri Jan 29 15:45:42 2021 +0000
@@ -375,6 +375,8 @@
     AllowTextPathChild,
     // The item is content created by an nsIAnonymousContentCreator frame.
     IsAnonymousContentCreatorContent,
+    // The item will be the rendered legend of a <fieldset>.
+    IsForRenderedLegend,
   };
 
   using ItemFlags = mozilla::EnumSet<ItemFlag>;
@@ -1086,7 +1088,8 @@
           mIsAllInline(false),
           mIsBlock(false),
           mIsPopup(false),
-          mIsLineParticipant(false) {
+          mIsLineParticipant(false),
+          mIsRenderedLegend(false) {
       MOZ_COUNT_CTOR(FrameConstructionItem);
     }
 
@@ -1160,6 +1163,8 @@
     bool mIsPopup : 1;
     // Whether this item should be treated as a line participant
     bool mIsLineParticipant : 1;
+    // Whether this item is the rendered legend of a <fieldset>
+    bool mIsRenderedLegend : 1;
 
    private:
     // Not allocated from the general heap - instead, use the new/Delete APIs